### PR TITLE
new protocol handler: stream-magnet

### PR DESCRIPTION
**Pull request by anonymlol**
Related to  #83 

Only tested on Windows.

### CODE REVIEW

File: renderer/main.js [hunk 1]
```diff
@@ -549,7 +549,7 @@ function onOpen (files) {
 function isTorrent (file) {
   var name = typeof file === 'string' ? file : file.name
   var isTorrentFile = path.extname(name).toLowerCase() === '.torrent'
-  var isMagnet = typeof file === 'string' && /^magnet:/.test(file)
+  var isMagnet = typeof file === 'string' && /^magnet:/.test(file) || /^stream-magnet:/.test(file)
```
Comment (feross) [inline renderer/main.js]: I would put parens around the magnet regexes for clarity -- I can never remember the order of ops.

File: main/handlers.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ function installDarwin () {
 
   // On OS X, only protocols that are listed in `Info.plist` can be set as the
   // default handler at runtime.
-  app.setAsDefaultProtocolClient('magnet')
+  app.setAsDefaultProtocolClient('stream-magnet')
```
Comment (feross) [inline main/handlers.js]: This shouldn't remove magnet, since we support both in Windows/Linux.

**Comment by feross** [comment]
LGTM, with a couple of exceptions inline. I'll fix them up after merge.

Thanks for the contribution!

**Comment by feross** [comment]
I also had to add support for `stream-magnet` to `parse-torrent`

File: renderer/main.js (hunk 1 above)
Comment (Anmo) [inline renderer/main.js]: This also could be like `var isMagnet = typeof file === 'string' && /^(stream-)?magnet:/.test(file)`

**Pull request merged**